src/sqlfluff/cli/commands.py

```diff
@@ -62,6 +64,24 @@ def filter(self, record: logging.LogRecord) -> bool:
         return True
 
 
+class StreamHandlerTqdm(logging.StreamHandler):
+    """Modified StreamHandler which takes care of writing within `tqdm` context.
+
+    It uses `tqdm` write which takes care of conflicting prints with progressbar.
+    Without it, there were left artifacts in DEBUG mode (not sure about another ones,
+    but probably would happen somewhere).
+    """
+
+    def emit(self, record: LogRecord) -> None:
+        """Behaves like original one except uses `tqdm` to write."""
+        try:
+            msg = self.format(record)
+            tqdm.write(msg, file=self.stream)
+            self.flush()
```

Comment:
Just curious -- what happens if you omit this line?

Reply:
Seems to be working as far I can tell, but it is in original `emit`, so I guess this is for some reason. So it safer just to put it there.